src/overlays/actors/ovl_Bg_Jya_Cobra/z_bg_jya_cobra.c

```diff
@@ -26,41 +52,628 @@ const ActorInit Bg_Jya_Cobra_InitVars = {
+    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_bg_jya_cobra.c", 966);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Cobra/func_80896918.s")
+    func_80094044(globalCtx->state.gfxCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Cobra/func_80896950.s")
+    if (sp72 == 0) {
+        sp64.x = this->dyna.actor.posRot.pos.x - 50.0f;
+        sp64.y = this->dyna.actor.posRot.pos.y;
+        sp64.z = this->dyna.actor.posRot.pos.z;
+        phi_a3 = &D_80897538;
+    } else {
+        phi_a3 = &this->dyna.actor.shape.rot;
+        if (sp72 == 2) {
+            sp64.x = this->dyna.actor.posRot.pos.x + 70.0f;
+            sp64.y = this->dyna.actor.posRot.pos.y;
+            sp64.z = this->dyna.actor.posRot.pos.z;
+            phi_a3 = &D_80897540;
+        } else {
+            Math_Vec3f_Copy(&sp64, &this->dyna.actor.posRot.pos);
+        }
+    }
+    func_800D1694(sp64.x, sp64.y, sp64.z, phi_a3);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Cobra/func_808969F8.s")
+    Matrix_Scale(D_80897548[sp72].x, D_80897548[sp72].y, D_80897548[sp72].z, MTXMODE_APPLY);
+    Matrix_Translate(0.0f, 0.0f, 40.0f, MTXMODE_APPLY);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Cobra/func_80896ABC.s")
+    gDPSetPrimColor(POLY_XLU_DISP++, 0, 0, 0, 0, 0, 120);
+    gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_bg_jya_cobra.c", 994),
+              G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+    gDPSetTextureImage(POLY_XLU_DISP++, G_IM_FMT_I, G_IM_SIZ_16b, 1, ALIGN16((s32)(&this->unk_194)));
+    gDPSetTile(POLY_XLU_DISP++, G_IM_FMT_I, G_IM_SIZ_16b, 0, 0x0000, G_TX_LOADTILE, 0, G_TX_NOMIRROR | G_TX_CLAMP,
+               G_TX_NOMASK, G_TX_NOLOD, G_TX_NOMIRROR | G_TX_CLAMP, G_TX_NOMASK, G_TX_NOLOD);
+    gDPLoadSync(POLY_XLU_DISP++);
+    temp = sp72;
+    gDPLoadBlock(POLY_XLU_DISP++, G_TX_LOADTILE, 0, 0, 2047, 256);
+    if (!temp) {}
+    gDPPipeSync(POLY_XLU_DISP++);
+    gDPSetTile(POLY_XLU_DISP++, G_IM_FMT_I, G_IM_SIZ_8b, 8, 0x0000, G_TX_RENDERTILE, 0, G_TX_NOMIRROR | G_TX_CLAMP,
+               G_TX_NOMASK, G_TX_NOLOD, G_TX_NOMIRROR | G_TX_CLAMP, G_TX_NOMASK, G_TX_NOLOD);
+    gDPSetTileSize(POLY_XLU_DISP++, G_TX_RENDERTILE, 0, 0, 0x00FC, 0x00FC);
+    gSPDisplayList(POLY_XLU_DISP++, D_808972B0);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Cobra/BgJyaCobra_Update.s")
+    CLOSE_DISPS(globalCtx->state.gfxCtx, "../z_bg_jya_cobra.c", 1006);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Cobra/func_80896CB4.s")
+void BgJyaCobra_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    BgJyaCobra* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Cobra/func_80896D78.s")
+    func_80896CB4(globalCtx);
+    Gfx_DrawDListOpa(globalCtx, &D_06010790);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Cobra/func_80896EE4.s")
+    if (0.0f < this->unk_18C) {
```

Comment:
```suggestion
    if (this->unk_18C > 0.0f) {
```
Usually we have variables on the left and values on the right if it matches